### ISSUE TITLE
Vector data SHM name can be too long on macOS

### ISSUE THREAD

**Issue by torokati44** [omnetpp/omnetpp#793]
- OMNeT++ version: 6.0pre6
 - OS: macOS (v10.14.6)

When querying vector data from chart scripts, the unique name generated for the SHM object used to pass the data between processes might be too long for the weird limitations of the POSIX implementation in macOS.
Originally reported by @screw.

```
py4j.protocol.Py4JJavaError: An error occurred while calling o2.getVectorsPickle.
: java.lang.RuntimeException: Error opening SHM file descriptor for '/vectordata-667963495733494-11-x': File name too long
at org.omnetpp.scave.engine.ScaveEngineJNI.createSharedMemory(Native Method)
at org.omnetpp.scave.engine.ScaveEngine.createSharedMemory(ScaveEngine.java:70)
at org.omnetpp.scave.python.ResultPicklingUtils.pickleXYArray(ResultPicklingUtils.java:63)
at org.omnetpp.scave.python.VectorResultsPickler.pickleVectorResult(VectorResultsPickler.java:34)
at org.omnetpp.scave.python.VectorResultsPickler.pickle(VectorResultsPickler.java:75)
at net.razorvine.pickle.Pickler.dispatch(Pickler.java:287)
```
See also: https://github.com/jackaudio/jack1/blob/c758cdf/libjack/shm.c#L700-L705

**Comment by torokati44** [comment]
A proposed fix is committed in the internal development repo.

**Comment by screw** [comment]
Is there release with this fix? (I tried to clone the repo but setenv warns that I am on MacOS and  ./configure does not detect Qt)

**Comment by screw** [comment]
@rhornig Thank you!

**Comment by torokati44** [comment]
@screw Can you confirm that this issue is solved for you in 6.0pre7?

Also, we (cross-)build all our releases inside a Docker container, which has all the right tools and libraries set up correctly. Take a look at the `releng/build-omnetpp-in-docker` script. (Some other ones in that folder are executed by the former, within the container.)
